controllers/etcdcopybackupstask/reconciler.go

```diff
@@ -283,6 +277,218 @@ func getConditionType(jobConditionType batchv1.JobConditionType) druidv1alpha1.C
+			MountPath: "/root/." + getVolumeNamePrefix(volumeMountPrefix) + "gcp/",
+		})
+	case druidutils.S3, druidutils.ABS, druidutils.Swift, druidutils.OCS, druidutils.OSS:
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      getVolumeNamePrefix(volumeMountPrefix) + "etcd-backup",
+			MountPath: "/root/" + getVolumeNamePrefix(volumeMountPrefix) + "etcd-backup/",
+		})
+	}
+	return
+}
+
+func getEnvVarFromValues(name, value string) corev1.EnvVar {
+	return corev1.EnvVar{
+		Name:  name,
+		Value: value,
+	}
+}
+
+func getProviderEnvs(store *druidv1alpha1.StoreSpec, storeProvider, prefix, volumePrefix string) []corev1.EnvVar {
+	env := []corev1.EnvVar{getEnvVarFromValues(prefix+"STORAGE_CONTAINER", *store.Container)}
+	switch storeProvider {
+	case druidutils.S3:
+		env = append(env, getEnvVarFromValues(prefix+"AWS_APPLICATION_CREDENTIALS", "/root/"+volumePrefix+"etcd-backup"))
+	case druidutils.ABS:
+		env = append(env, getEnvVarFromValues(prefix+"AZURE_APPLICATION_CREDENTIALS", "/root/"+volumePrefix+"etcd-backup"))
+	case druidutils.GCS:
+		env = append(env, getEnvVarFromValues(prefix+"GOOGLE_APPLICATION_CREDENTIALS", "/root/."+volumePrefix+"gcp/serviceaccount.json"))
+	case druidutils.Swift:
+		env = append(env, getEnvVarFromValues(prefix+"OPENSTACK_APPLICATION_CREDENTIALS", "/root/"+volumePrefix+"etcd-backup"))
+	case druidutils.OCS:
+		env = append(env, getEnvVarFromValues(prefix+"OPENSHIFT_APPLICATION_CREDENTIALS", "/root/"+volumePrefix+"etcd-backup"))
+	case druidutils.OSS:
+		env = append(env, getEnvVarFromValues(prefix+"ALICLOUD_APPLICATION_CREDENTIALS", "/root/"+volumePrefix+"etcd-backup"))
+	}
+	return env
+}
+
+func setEtcdBackupJobCommands(task *druidv1alpha1.EtcdCopyBackupsTask, command *[]string) error {
+	provider, err := druidutils.StorageProviderFromInfraProvider(task.Spec.TargetStore.Provider)
+	if err != nil {
+		return err
+	}
+
+	if len(provider) > 0 {
+		*command = append(*command, "--storage-provider="+provider)
+	}
+
+	if len(task.Spec.TargetStore.Prefix) > 0 {
+		*command = append(*command, "--store-prefix="+task.Spec.TargetStore.Prefix)
+	}
+
+	if task.Spec.TargetStore.Container != nil && len(*task.Spec.TargetStore.Container) > 0 {
+		*command = append(*command, "--store-container="+*task.Spec.TargetStore.Container)
+	}
+
+	provider, err = druidutils.StorageProviderFromInfraProvider(task.Spec.SourceStore.Provider)
+	if err != nil {
+		return err
+	}
+
+	if len(provider) > 0 {
+		*command = append(*command, "--source-storage-provider="+provider)
+	}
+
+	if len(task.Spec.SourceStore.Prefix) > 0 {
+		*command = append(*command, "--source-store-prefix="+task.Spec.SourceStore.Prefix)
+	}
+
+	if task.Spec.SourceStore.Container != nil && len(*task.Spec.SourceStore.Container) > 0 {
+		*command = append(*command, "--source-store-container="+*task.Spec.SourceStore.Container)
+	}
+
+	if task.Spec.MaxBackupAge != nil && *task.Spec.MaxBackupAge != 0 {
```

Comment:
there is no check on the value of maxbackupage in the chart. Here you are only setting if the value is not 0. What is consequence if the value is set to 0?